mesonbuild/modules/external_project.py

```diff
@@ -116,6 +116,19 @@ def _configure(self, state: 'ModuleState') -> None:
             configure_path = Path(self.src_dir, self.configure_command)
             configure_prog = state.find_program(configure_path.as_posix())
             configure_cmd = configure_prog.get_command()
+            # On Cygwin, MSYS2 and GitBash the configure command should be
+            # converted to unix style path by cygpath command,
+            # because the colon in the drive letter breaks many configure scripts.
+            # Do nothing on other environment where cygpath is not available.
+            if (
+                configure_path.drive
+                and len(configure_cmd) >= 2
+                and configure_cmd[-1] == configure_path.as_posix()
+            ):
+                cygpath = state.find_program('cygpath', required=False)
```

Comment:
shutil.which can find it just like state.find_program(), with the distinction that state.find_program() is intended to be plumbed through machine files in the event that users need to control "exactly which version". I don't think the find_program machinery does anything useful here?